text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+- Start Date: 2023-09-10
+- RFC PR: [rust-lang/rfcs#3488](https://github.com/rust-lang/rfcs/pull/3488)
+
+# Summary
+
+The first part of this RFC contains a summary of the "categories" and "keywords"
+concepts, their combinations, their pros and cons, their usage statistics and
+how they are used in other ecosystems.
+
+The second part proposes merging the concepts of "categories" and "keywords" on
+crates.io into a single concept, achieving the best of both worlds with a list
+of recommended keywords.
+
+
+# History
+
+- June 2014:
+  - [First commit to crates.io](https://github.com/rust-lang/crates.io/commit/54cfc8d)
+- Oct. 2014: 
+  - [Keywords are added to crates.io](https://github.com/rust-lang/crates.io/commit/8f03b72)
+- Nov. 2016:
+  - [Categories are added to crates.io](https://github.com/rust-lang/crates.io/pull/473)
+  - [Discussion on the initial list of categories is started](https://github.com/rust-lang/crates.io/pull/488)
+    - The initial list was adopted from https://github.com/rust-unofficial/awesome-rust
+    - A couple of comments suggest using a list of "official" keywords instead
+      of introducing a parallel concept of categories (https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195
+      and https://github.com/rust-lang/crates.io/pull/488#issuecomment-265623340)
+
+
+# Statistics
+
+The following numbers have been extracted from a recent
+[crates.io database dump](https://crates.io/data-access#database-dumps):
+
+- 40,649 crates have `categories`
+  - 3,464 of those crates have only `categories`, and no `keywords`
+  - 3,622 of those crates have only a single category
+- 61,216 crates have `keywords`
+  - 24,031 of those crates have only `keywords`, and no `categories`
+  - 4,694 of those crates have only a single keyword
+- 37,185 crates have both `categories` and `keywords`
+- 56,694 crates have neither `categories`, nor `keywords`
+
+
+# Keywords vs. Categories
+
+**Keywords** (aka. tags) describe an open set of names to categorize packages.
+The set being "open" means that users can use arbitrary keywords without having
+to choose from a fixed list of available names.
+
+**Categories** describe a closed set of names to categorize packages. The set 
+being "closed" means that users have to choose from a fixed list of available
+categories, and can't easily add new ones.
```

Comment:
Extremely minor nit: would be worth keeping the order of "categories and keywords" when describing things just to avoid confusion when reading.

```suggestion
# Categories vs. Keywords

**Categories** describe a closed set of names to categorize packages. The set 
being "closed" means that users have to choose from a fixed list of available
categories, and can't easily add new ones.

**Keywords** (aka. tags) describe an open set of names to categorize packages.
The set being "open" means that users can use arbitrary keywords without having
to choose from a fixed list of available names.
```

Reply:
makes sense, thanks :)